app/src/test/java/org/mozilla/fenix/AppRequestInterceptorTest.kt

```diff
@@ -0,0 +1,113 @@
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class AppRequestInterceptorTest {
+
+    private lateinit var interceptor: RequestInterceptor
+
+    @Before
+    fun setUp() {
+        interceptor = AppRequestInterceptor(testContext)
+    }
+
+    @Test
+    fun `onErrorRequest results in correct error page for low risk level error`() {
+        setOf(
+            ErrorType.UNKNOWN,
+            ErrorType.ERROR_NET_INTERRUPT,
+            ErrorType.ERROR_NET_TIMEOUT,
+            ErrorType.ERROR_CONNECTION_REFUSED,
+            ErrorType.ERROR_UNKNOWN_SOCKET_TYPE,
+            ErrorType.ERROR_REDIRECT_LOOP,
+            ErrorType.ERROR_OFFLINE,
+            ErrorType.ERROR_NET_RESET,
+            ErrorType.ERROR_UNSAFE_CONTENT_TYPE,
+            ErrorType.ERROR_CORRUPTED_CONTENT,
+            ErrorType.ERROR_CONTENT_CRASHED,
+            ErrorType.ERROR_INVALID_CONTENT_ENCODING,
+            ErrorType.ERROR_UNKNOWN_HOST,
+            ErrorType.ERROR_MALFORMED_URI,
+            ErrorType.ERROR_FILE_NOT_FOUND,
+            ErrorType.ERROR_FILE_ACCESS_DENIED,
+            ErrorType.ERROR_PROXY_CONNECTION_REFUSED,
+            ErrorType.ERROR_UNKNOWN_PROXY_HOST,
+            ErrorType.ERROR_UNKNOWN_PROTOCOL
+        ).forEach { error ->
+            val (actualPage, expectedPage) = createErrorPages(
+                error = error,
+                html = R.raw.low_risk_error_pages,
+                css = R.raw.low_and_medium_risk_error_style
+            )
+
+            assertThat(actualPage).isEqualTo(expectedPage)
+        }
+    }
+
+    @Test
+    fun `onErrorRequest results in correct error page for medium risk level error`() {
+        setOf(
+            ErrorType.ERROR_SECURITY_BAD_CERT,
+            ErrorType.ERROR_SECURITY_SSL,
+            ErrorType.ERROR_PORT_BLOCKED
+        ).forEach { error ->
+            val (actualPage, expectedPage) = createErrorPages(
+                error = error,
+                html = R.raw.medium_and_high_risk_error_pages,
+                css = R.raw.low_and_medium_risk_error_style
+            )
+
+            assertThat(actualPage).isEqualTo(expectedPage)
+        }
+    }
+
+    @Test
+    fun `onErrorRequest results in correct error page for high risk level error`() {
+        setOf(
+            ErrorType.ERROR_SAFEBROWSING_HARMFUL_URI,
+            ErrorType.ERROR_SAFEBROWSING_MALWARE_URI,
+            ErrorType.ERROR_SAFEBROWSING_PHISHING_URI,
+            ErrorType.ERROR_SAFEBROWSING_UNWANTED_URI
+        ).forEach { error ->
+            val (actualPage, expectedPage) = createErrorPages(
+                error = error,
+                html = R.raw.medium_and_high_risk_error_pages,
+                css = R.raw.high_risk_error_style
+            )
+
+            assertThat(actualPage).isEqualTo(expectedPage)
+        }
+    }
+
+    private fun createErrorPages(error: ErrorType, @RawRes html: Int, @RawRes css: Int): Pair<String, String> {
```

Comment:
There's nothing wrong with this code, but in the interest of keeping the test as simple to read as possible would you mind breaking it into two functions?  I think it's worth the minor duplication.